app/controllers/two_factor_authentication/personal_key_verification_controller.rb

```diff
@@ -24,6 +25,12 @@ def create
 
     private
 
+    def check_personal_key_enabled
+      return if TwoFactorAuthentication::PersonalKeyPolicy.new(current_user).enabled?
+
+      redirect_to two_factor_options_url
```

Comment:
It looks like this line does not have coverage. I think it might make sense to add a controller spec to make sure that IAL2 users and users w/o personal keys get redirected.